### DISCUSSION TITLE
プロフィールページにその人の記事がLGTM順に並ぶビューがほしい

### DISCUSSION THREAD

**Discussion by JunichiIto** [increments/qiita-discussions#219] [要望・アイデア・不具合]
表題の通り、プロフィールページにその人の記事がLGTM順に並ぶビューがほしいです。
具体的なユースケースとしては自分の話になるのですが、「今までたくさん記事を書いてきたけど、人気が出た記事ってどれだっけ？」というのをふと調べたくなったからです。

自分の記事だけではなく、他の人のプロフィールを見たときもそういうビューがあると、「あ、この人はこんな人気記事を書いてたのか」というのがわかって嬉しいことがあるかもしれません。

「実は今でもこういうふうにやれば実現できますよ」という裏技（？）があったら教えてください。
ただ、理想としては裏技ではなくプロフィールページでぱっとわかるようになってる方が嬉しいです🙏

**Comment by munierujp** [comment]
>「実は今でもこういうふうにやれば実現できますよ」という裏技（？）があったら教えてください。

自分は、ピックアップ記事にLGTM上位3件を設定しています（めったに変動しません）。
また、記事検索で`user:<username>`のように検索してLGTM数順にソートすればその人の記事がLGTM数順に表示されます。
e.g. https://qiita.com/search?q=user%3Amunieru_jp&sort=like
しかしながらこれは結構面倒なので、プロフィールページで同じことができるといいですね。

**Comment by yoshi389111** [comment]
検索画面で、ユーザー名で絞り込んで（例えば `user:jnchito` ）、ソート順を「LGTM数順」にすると、それっぽいことは可能だと思います。ちょっと面倒ですが。

個人的には、ユーザーのプロフィール画面の記事一覧の右上あたりに、検索アイコン（虫眼鏡とか？）があって、それをクリックしたら、ユーザー名で簡単に絞り込んで検索できるようになっていれば、それでも十分使えそうな気がしています。

**Comment by JunichiIto** [comment]
@munierujp @yoshi389111 情報ありがとうございます！なるほど、こうやればできるんですね〜。
https://qiita.com/search?sort=like&q=user%3Ajnchito

回避策が見つかったので優先度は低めでも良いと思いますが、ワンクリックで飛べるようなショートカットがプロフィールページにあると嬉しいですね。

**Reply by gilly135** [reply]
@JunichiIto 
ご意見ありがとうございました。

>ワンクリックで飛べるようなショートカットがプロフィールページにあると嬉しいですね。

時間が空いてしまいましたがこちらのご意見に対し、ユーザーページから、閲覧中のユーザーを検索条件に指定した検索結果ページに遷移できるようにしました！

詳細はリリースノートをご確認ください。
https://qiita.com/release-notes/2024-07-24

ご利用いただき、またご要望などございましたらお知らせいただけると嬉しいです 🙏

**Reply by JunichiIto** [reply]
@gilly135 確認しました。対応ありがとうございます！
とても便利になりました 🙌

**Comment by tanaka-asami415** [comment]
@JunichiIto @munierujp @yoshi389111 

ご意見ありがとうございます🙏
マイページを改善していきたいとは思っているので、その際に今回頂いた内容も検討させていただければと思います！
お手数をおかけしてすみませんが、いまのところは検索画面等で対応いただければ幸いです🙇

**Reply by lamrongol** [reply]
単純なLGTM順だとシリーズ物の記事がバラバラになってしまい、また、LGTM数に関わらずオススメの記事などもあるので、できれば手動で記事を並べ替える機能をつけて欲しいです。

**Reply by tanaka-asami415** [reply]
@lamrongol 
ご意見ありがとうございます🙏
> できれば手動で記事を並べ替える機能をつけて欲しいです。

以下のご希望があり、手動で並べ替えをされたいのではないかと理解したのですが
- シリーズ物を連続で表示させたい
- LGTM数に関わらずオススメの記事を見せたい

現在、自分のマイページでおすすめ記事を表現する機能（ピックアップ記事）がありますが
ここでピックアップできる記事数が増えれば、ご要望は叶えられそうでしょうか？
ご要望と違うようでしたら、もう少し詳しく教えていただけると嬉しいです🙏🙏

**Reply by lamrongol** [reply]
確かにピックアップ記事数の増加でも要望は叶えられそうです。ただ現在は記事の順番までは指定できないので、ピックアップ記事数を無制限にして順番も指定できるようにしていただけると嬉しいです。

**Reply by tanaka-asami415** [reply]
@lamrongol 
詳細を教えていただきありがとうございます！🙏

順番に関しては、現時点の対応策としましては、お手数なのですが、一度ピックアップから外して
再度登録いただくと、下に順番に追加されていきますので、こちらの方法で順番を変えることは可能です🙏
ただ、いい体験とは言えないため、マイページ改善の際に合わせて改善したいと思います💪💪

**Comment by tsubolabo** [comment]
> https://qiita.com/release-notes/2024-07-24

　この方法の検索結果で「いいね数順」を選択しても、ソート結果が完全にはなりません（順番がやや乱れます）。「ストック数順」の方は正しい順番で表示されます。このページに限っては、「いいね数順」を別の評価（例えば「Contribution順」）とでも読み替える必要があるのでしょうか？　しかし、変則的な解釈を求められると混乱します。私は「いいね数順」で正しく表示されることを希望します。

　参考までに、自分のページで「いいね数順」で検索した結果の記事の「いいね数 / ストック数」を以下に順番に示しています。高評価の方の順番乱れは、「Contribution順」と言われれば何となく「そうかな」と納得できますが、低評価の方の順番乱れについては、どうにも意味が分かり兼ねます。

23  / 11
17 /  11
19 /  1
11 /  4
 9  / 1
 8  / 2
 6  / 2
 6  / 2
 5  / 1
 4  / 1
 4  / 7
 3  / 1
 3  / 0
 3  / 0
 2  / 0
 2  / 0
 2  / 0
 2  / 2
 2  / 0
 3  / 2
 1  / 0
 1  / 0
 2  / 0
 0  / 0
 0  / 0
 3  / 0
 0  / 2
 0  / 0
 1  / 0
 0  / 0
 0  / 1

**Reply by tanaka-asami415** [reply]
@tsubolabo 
ご不便をおかけして申し訳ございません。
いいね数が検索結果に反映されるまでに時差があり、上記のような現象が起きております。
修正する予定ですので、お待ちいただけると幸いです🙇
